core/sql/arkcmp/CompException.cpp

```diff
@@ -113,11 +114,21 @@ FatalException::FatalException(const char *msg,
   else {
     msg_[0] = 0;
   }
+  if(stackTrace) {
+    strncpy(stackTrace_, stackTrace, sizeof(stackTrace_));
+    stackTrace_[sizeof(stackTrace_)-1] = 0;
```

Comment:
Though sizeof(stackTrace_) should work fine in this case  you are implicitly assuming that it is not a pointer